independent-projects/tools/devtools-common/src/main/java/io/quarkus/registry/model/Release.java

```diff
@@ -15,6 +16,9 @@
     @JsonProperty("quarkus-core")
     String getQuarkusCore();
 
+    @JsonProperty("compatible-quarkus-core")
+    List<String> getCompatibleQuarkusCore();
```

Comment:
So how would this work? Should the JSON data corresponding to this field be updated everytime a new release is out?

Reply:
This is surely a great topic for a big discussion. Which probably shouldn't be in the form of these comments. It's nevertheless a relevant question for the change. So here is an idea (which I also mentioned in one of my emails to the dev list awhile ago now).
We definitely need this info (Quarkus core compatibility) in one way or another. W/o it we'd be either suggesting potentially incompatible extension releases to projects or suggesting only those that were built with the Quarkus core version matching the one used in the user project ignoring the rest of the ecosystem that is still compatible.
At this point, my not especially deep thought out idea is to support promoting the latest extension releases to the catalogs based on newer Quarkus versions than the one they were originally built with.
For example, let's say there is a registered extension A with release 0.1 built with Quarkus core 1.11.0.Final. When Quarkus 1.12.0.Final is released we/CI would check whether A-0.1 is compatible with it. If it is, we add 1.12.0.Final to `compatible-quarkus-core` and A-0.1 appears in the catalogs based on 1.12.0.Final.
Then if Quarkus 1.13.0.Final is out we follow the same process. But we follow the same process only if A-0.1 actually was compatible with 1.12.0.Final. If A-0.1 didn't appear to be compatible with 1.12.0.Final we, by default, exclude it from further compatibility checks with new Quarkus releases. This is to avoid wasting CI resources on potentially not actively developed or abandoned projects. The compatibility info, btw, can also be added any time "manually" in case some extension catches up and needs to be included into the catalogs.
So, basically, when a new Quarkus release is out we would be going through all the latest non-platform extension releases looking for those compatible with the previous Quarkus release and checking whether they are still compatible with the newly released Quarkus core.
As soon as A 0.2 is out, we, by default, stop checking A-0.1 for compatibility with the new Quarkus releases. A-0.1 compatibility could still be adjusted "manually" of course.
A similar case could happen in case A 0.3 is out and somebody wants to indicate that it's compatible with Quarkus 1.11.0.Final.
In both of these cases, we'll end with multiple releases of the same extension (A) that are compatible with the same version of Quarkus (e.g. 1.11.0.Final). Which creates a choice when generating catalog for that version of Quarkus. I guess we would pick the latest release of A as the preferred one by default.